test/test_jit.py

```diff
@@ -1236,124 +1236,164 @@ def observer(self, x, name):
-        # is last node. Constant nodes correspond to params for the
-        # quantization nodes
-        FileCheck().check("quantize_linear").check_next("dequantize") \
-                   .check("_convolution").check_next("Constant") \
-                   .check_next("Constant").check_next("quantize_linear") \
-                   .check_next("dequantize").run(str(trace.graph))
-        FileCheck().check("relu").check_next("Constant") \
-                   .check_next("Constant").check_next("quantize_linear") \
-                   .check_next("dequantize").check_next("return") \
-                   .run(str(trace.graph))
-
-    def test_insert_quantdequant_single_qnode(self):
-        class testModule(torch.jit.ScriptModule):
-            def __init__(self):
-                super(testModule, self).__init__()
-                self.conv1 = nn.Conv2d(1, 20, 5, 1)
+        # Helper function to run quantdequant pass for all tests
+        def test_helper_run_quantdequant(trace, x1):
+            activationObj = TestGenQParam('activation')
+            weightObj = TestGenQParam('param')
+            qconfig_dict = {'__main__': {'activation': activationObj,
+                                         'param': weightObj}
+                            }
+            torch._C._jit_pass_insert_observers(trace, qconfig_dict)
+            trace.forward(x1)
+            # qparam_dict
+            allqparam_dict = {}
+            allqparam_dict.update(activationObj.qparam_dict)
+            allqparam_dict.update(weightObj.qparam_dict)
+            torch._C._jit_pass_insert_quantdequant(trace.graph, allqparam_dict)
+
+        def test_insert_quantdequant_consecutive_qnodes_script():
+            class testModule(torch.jit.ScriptModule):
+                def __init__(self):
+                    super(testModule, self).__init__()
+                    self.conv1 = nn.Conv2d(1, 1, 1, 1)
 
-            @torch.jit.script_method
-            def forward(self, x):
-                x = self.conv1(x)
-                x1 = torch.add(x, 1)
-                return x1
+                @torch.jit.script_method
+                def forward(self, x):
+                    x = F.relu(self.conv1(x))
+                    return x
 
-        trace = testModule()
+            trace = testModule()
+
+            # Constant Propagation step is performed because this pass is intended
+            # to insert quant-dequant nodes for quantizable tensors. The type analysis
+            # happens as part of this jit pass
+            torch._C._jit_pass_constant_propagation(trace.graph)
+
+            test_helper_run_quantdequant(trace, x1)
+
+            # We expect to see quant-dequant node before and after
+            # both conv and relu nodes and at external output since relu
+            # is last node. Constant nodes correspond to params for the
+            # quantization nodes
+            FileCheck().check("quantize_linear").check_next("dequantize") \
```

Comment:
Please use a new style of writing tests, an example can be found in `test_recursive_cse` test, a documentation is available here: https://github.com/pytorch/pytorch/blob/master/test/HowToWriteTestsUsingFileCheck.md. It allows to make the intention of FileCheck checks clearer.